test/segments/stack_project.spec

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env zsh
+#vim:ft=zsh ts=2 sw=2 sts=2 et fenc=utf-8
+
+# Required for shunit2 to run correctly
+setopt shwordsplit
+SHUNIT_PARENT=$0
+
+function setUp() {
+  export TERM="xterm-256color"
+  # Load Powerlevel9k
+  source powerlevel9k.zsh-theme
+
```

Comment:
Is there a reason why `source segments/stack_project.p9k` is done in every test, instead of here?

Reply:
Yes. I tried putting that in `setUp()` as you suggested at Gitter, but then the mock icon assigned with `local P9K_STACK_PROJECT_ICON='x'` (at lines 51, 66, 80 and 93) is not loaded correctly, making the assertions fail. Is there another way to do it?

Reply:
Yep. There are two ways to handle that:
1. Set the icon _before_ you source the segment. This would be then set for all tests.
2. Do not overwrite the icon. You could just copy/paste the glyph into the test. The test then should be resilient against different `P9K_MODE`s..

Reply:
Regarding option 2, with the glyph I guess you mean the generic one? Because it would vary if NerdFonts are installed wouldn't it?

Reply:
Right. That is why the test should set a `P9K_MODE` itself (and backing up the old one before), so that the test always runs under the same conditions.